aws/rust-runtime/aws-types/src/profile/parse.rs

```diff
@@ -0,0 +1,333 @@
+            .strip_suffix(']')
+            .ok_or_else(|| self.emit_error("Profile definition must end with ']'"))?;
+        if !self.data.contains_key(profile_name) {
+            self.data.insert(profile_name, Default::default());
+        }
+        self.state = State::ReadingProfile {
+            profile: profile_name,
+            property: None,
+            is_subproperty: false,
+        };
+        Ok(())
+    }
+}
+
+/// Error encountered while parsing a property
+#[derive(Debug, Eq, PartialEq)]
+enum PropertyError {
+    NoEquals,
+    NoName,
+}
+
+impl PropertyError {
+    fn into_error(self, ctx: &str, location: Location) -> ProfileParseError {
+        let mut ctx = ctx.to_string();
+        match self {
+            PropertyError::NoName => {
+                ctx.get_mut(0..1).unwrap().make_ascii_uppercase();
+                ProfileParseError {
+                    location,
+                    message: format!("{} did not have a name", ctx),
+                }
+            }
+            PropertyError::NoEquals => ProfileParseError {
+                location,
+                message: format!("Expected an '=' sign defining a {}", ctx),
+            },
+        }
+    }
+}
+
+/// Parse a property line into a key-value pair
+fn parse_property_line(line: &str) -> Result<(&str, &str), PropertyError> {
+    let line = prepare_line(line, true);
+    let (k, v) = line.split_once('=').ok_or(PropertyError::NoEquals)?;
+    let k = k.trim_matches(WHITESPACE);
+    let v = v.trim_matches(WHITESPACE);
+    if k.is_empty() {
+        return Err(PropertyError::NoName);
+    }
+    Ok((k, v))
+}
+
+/// Prepare a line for parsing
+///
+/// This function preprocesses a line to simplify parsing:
+/// 1. Strip leading and trailing whitespace
+/// 2. Remove trailing comments
+///
+/// Depending on context, comment characters may need to be preceded by whitespace to be considered
+/// comments.
+fn prepare_line(line: &str, comments_need_whitespace: bool) -> &str {
+    let line = line.trim_matches(WHITESPACE);
+    let mut prev_char_whitespace = false;
+    let mut comment_idx = None;
+    for (idx, chr) in line.char_indices() {
+        if (COMMENT.contains(&chr)) && (prev_char_whitespace || !comments_need_whitespace) {
+            comment_idx = Some(idx);
+            break;
+        }
+        prev_char_whitespace = chr.is_whitespace();
+    }
+    comment_idx
+        .map(|idx| &line[..idx])
+        .unwrap_or(&line)
+        // trimming the comment might result in more whitespace that needs to be handled
+        .trim_matches(WHITESPACE)
+}
+
+#[cfg(test)]
+mod test {
+    use super::{parse_profile_file, prepare_line, Location};
+    use crate::profile::parse::{parse_property_line, PropertyError};
+    use crate::profile::source::File;
+
+    // most test cases covered by the JSON test suite
+
+    #[test]
+    fn property_parsing() {
+        assert_eq!(parse_property_line("a = b"), Ok(("a", "b")));
```

Comment:
Should also have a test for `a=b`, and also for `a=`.

Is it possible for the value to have an `=` in it? If so, there should be a test for that (i.e., `something_base64=aGVsbG8gZW50aHVzaWFzdGljIHJlYWRlcg==`).